packages/wp-now/src/wp-now.ts

```diff
@@ -0,0 +1,299 @@
+import fs from 'fs-extra';
+import crypto from 'crypto';
+import { NodePHP } from '@php-wasm/node';
+import { SupportedPHPVersion } from '@php-wasm/universal';
+import path from 'path';
+import {
+	SQLITE_FILENAME,
+	SQLITE_PATH,
+	WORDPRESS_ZIPS_PATH,
+	WP_NOW_PATH,
+} from './constants';
+import { downloadSqlite, downloadWordPress } from './download';
+import { portFinder } from './port-finder';
+
+type WPNowMode = 'plugin' | 'theme' | 'core' | 'index' | 'auto';
+export interface WPNowOptions {
+	phpVersion?: SupportedPHPVersion;
+	documentRoot?: string;
+	absoluteUrl?: string;
+	mode?: WPNowMode;
+	projectPath?: string;
+	wpContentPath?: string;
+}
+
+const DEFAULT_OPTIONS: WPNowOptions = {
+	phpVersion: '8.0',
+	documentRoot: '/var/www/html',
+	mode: 'auto',
+};
+async function getAbsoluteURL() {
+	const port = await portFinder.getOpenPort();
+	return `http://127.0.0.1:${port}`;
+}
+
+function seemsLikeAPHPFile(path) {
+	return path.endsWith('.php') || path.includes('.php/');
+}
+
+export default class WPNow {
+	php: NodePHP;
+	options: WPNowOptions = DEFAULT_OPTIONS;
+
+	static async create(options: WPNowOptions = {}): Promise<WPNow> {
+		const instance = new WPNow();
+		const absoluteUrl = await getAbsoluteURL();
+		const projectPath = options.projectPath || process.cwd();
+		const wpContentPath = this.#getWpContentHomePath(projectPath);
+		const mode = this.#inferMode(projectPath);
+		await instance.#setup({
+			absoluteUrl,
+			projectPath,
+			wpContentPath,
+			mode,
+			...options,
+		});
+		return instance;
+	}
+
+	updateFile = (path, callback) => {
+		this.php.writeFile(path, callback(this.php.readFileAsText(path)));
+	};
+
+	async #setup(options: WPNowOptions = {}) {
```

Comment:
are we gaining much through the private methods? are we worried someone will load these modules and call the functions?

Reply:
My reasoning for using private methods is to simplify the options for other devs, exposing only what's necessary.
If someone wants to use another method, I'll happily change it to public.